### PR TITLE
[4.2] Enable/disable 'Upload&Install' button on the basis of checkbox value

### PR DESCRIPTION
**Pull request by Shubhamverma2796**
Pull Request for Issue #36274 and #35055 .

### Summary of Changes
Added disabled state to the "UploadAndInstall" button.

### Testing Instructions

1.System-->Update-->Joomla.
2. Click on "Update your site by manually uploading the update package".
3. Button and checkbox are disabled initially.
4. Upload a file greater than 40 MB, checkbox and button are still disabled.
5. upload a file less than 40 MB, the checkbox is enabled. Check in the box, the button gets enabled.
6. again upload a file greater than 40 MB, checkbox and button get disabled again.

### Expected result AFTER applying this Pull Request

https://user-images.githubusercontent.com/54266411/151861745-104601f0-9a7c-4ef7-8479-108c6b4d1f2d.mp4

### CODE REVIEW

File: administrator/components/com_joomlaupdate/tmpl/upload/default.php [hunk 1]
```diff
@@ -84,13 +84,13 @@
 	</div>
 
 	<div class="form-check mb-3">
-		<input class="form-check-input me-2" type="checkbox" value="" id="joomlaupdate-confirm-backup">
+		<input class="form-check-input me-2" type="checkbox" onchange="document.getElementById('uploadButton').disabled = !this.checked;" value="" id="joomlaupdate-confirm-backup">
```
Comment (dgrammatiko) [inline administrator/components/com_joomlaupdate/tmpl/upload/default.php]: Please no inline script tags, it's 2022...

Comment (Shubhamverma2796) [reply administrator/components/com_joomlaupdate/tmpl/upload/default.php]: I thought this one is a minor change so I did. Can you please tell me if what's the ideal way of doing this? I'm supposed to create a function for the disabled state. Right?

Comment (dgrammatiko) [reply administrator/components/com_joomlaupdate/tmpl/upload/default.php]: The layout that you're editing has a script `->useScript('com_joomlaupdate.default')`, so do the changes in the js file

also fyi the reason is not the size but primarily this kind of coding will fail if someone enables CSP strict mode

Comment (Shubhamverma2796) [reply administrator/components/com_joomlaupdate/tmpl/upload/default.php]: I've removed the inline js.

**Comment by pritam825** [comment]
I have tested this item :white_check_mark: successfully on a3ce59d2a613a79c81e86082945dfdcd9f258274<br /><br />Working as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

File: administrator/components/com_joomlaupdate/tmpl/upload/default.php (hunk 1 above)
Comment (Shubhamverma2796) [reply administrator/components/com_joomlaupdate/tmpl/upload/default.php]: Can you please review the code again?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3ea1dc7785eae7238630a0bf1088295a1ba2fcb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by bembelimen** [comment]
After discussing with @Shubhamverma2796 we agreed, that the flow should also include the file input field, too. So a un-disable chain: field => checkbox => button

Thanks for the PR @Shubhamverma2796 and thanks for continuing working on it!

**Comment by bembelimen** [comment]
Removed RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by Aman21591** [comment]
I have tested this item :white_check_mark: successfully on f882b1d5b8dcd8e4e8992219b55767d7eb781b0e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by ditsuke** [comment]
I have tested this item :white_check_mark: successfully on f882b1d5b8dcd8e4e8992219b55767d7eb781b0e<br /><br />For any future testers, please note that you'll need to rebuild the JS assets for com_joomlaupdate even if you apply the patch via Joomla! Patch Tester. I had to run `npm run build:js  -- build/media_source/com_joomlaupdate` before I could reproduce the changes successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36860">issues.joomla.org/tracker/joomla-cms/36860</a>.</sub>

**Comment by bembelimen** [comment]
Thanks for this very nice enhancement, I rebased it to 4.2 as it's a new thing.

**Comment by Quy** [comment]
@Shubhamverma2796 Please consider doing the same in a separate PR for the following. Thanks.

update-joomla

**Comment by Shubhamverma2796** [comment]
> @Shubhamverma2796 Please consider doing the same in a separate PR for the following. Thanks.
> 
> update-joomla

I am not getting this screen.
Screenshot (352)

**Comment by Shubhamverma2796** [comment]
> > @Shubhamverma2796 Please consider doing the same in a separate PR for the following. Thanks.
> > update-joomla
> 
> I am not getting this screen. Screenshot (352)

I got the screen. Should I open a new issue or directly open a pr because #35055 is closed.

**Comment by Quy** [comment]
@Shubhamverma2796 Please create a new PR to disable the `Start update` button until the checkbox is checked.

**Comment by Shubhamverma2796** [comment]
> @Shubhamverma2796 Please create a new PR to disable the `Start update` button until the checkbox is checked.

Sure. I will do it ASAP.

**Comment by roland-d** [comment]
Thank you everybody :)

**Pull request merged**